modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java

```diff
@@ -1062,45 +1086,96 @@ private Node getLeaseholderNodeForPartition(Node node, int partId) {
                 .get();
     }
 
-    private static Set<Assignment> getPartitionClusterNodes(Node node, int partNum) {
-        return getPartitionClusterNodes(node, TABLE_NAME, partNum);
+    private static Set<Assignment> getPartitionStableAssignments(Node node, int partNum) {
+        return getPartitionStableAssignments(node, TABLE_NAME, partNum);
     }
 
-    private static Set<Assignment> getPartitionClusterNodes(Node node, String tableName, int partNum) {
-        return Optional.ofNullable(getTableId(node, tableName))
-                .map(tableId -> stablePartitionAssignments(node.metaStorageManager, tableId, partNum).join())
+    private static Set<Assignment> getPartitionStableAssignments(Node node, String tableName, int partNum) {
+        int zoneId = getZoneIdByTableNameStrict(node.catalogManager, tableName, node.hybridClock.nowLong());
```

Comment:
This pattern is copied many times, can we extract this somehow?

Reply:
Extracted to `TestRebalanceUtil` class.